tests/integration/vlan_test.py

```diff
@@ -35,6 +37,47 @@ def test_add_and_remove_vlan(eth1_up):
     assert not current_state[INTERFACES]
 
 
+def test_add_down_remove_vlan(eth1_up):
+    up = yaml.load("""
+interfaces:
+- name: eth1.101
+  type: vlan
+  state: up
+  vlan:
+    base-iface: eth1
+    id: 101
+""")
+    down = yaml.load("""
+interfaces:
+- name: eth1.101
+  type: vlan
+  state: down
+  vlan:
+    base-iface: eth1
+    id: 101
+""")
+
+    absent = yaml.load("""
+interfaces:
+- name: eth1.101
+  type: vlan
+  state: absent
+  vlan:
+    base-iface: eth1
+    id: 101
+""")
+
+    netapplier.apply(up)
```

Comment:
Please use a contextmanager in order to properly cleanup even on failure (of the apply or assert)

Reply:
changed it, this will hopefully show that absent fails to clean up ;-)